e2e/scripts/run_tests.sh

```diff
@@ -0,0 +1,169 @@
+    exit $2
+}
+
+build_images() {
+    echo "Building images..."
+    docker build -t soluto/oidc-server-mock '../src'
+    docker build -t soluto/e2e '.'
+}
+
+load_images() {
+    echo "Loading images to cluster..."
+    kind load docker-image --name "$CLUSTER_NAME" soluto/oidc-server-mock
+    kind load docker-image --name "$CLUSTER_NAME" soluto/e2e
+}
+
+prepare_environment() {
+    echo "Configuring kubectl..."
+    local oldKUBECONFIG="$KUBECONFIG"
+    KUBECONFIG="$(kind get kubeconfig-path --name=$CLUSTER_NAME)"
+    export KUBECONFIG
+
+    echo "Creating kubernetes objects..."
+
+    # oidc-service
+    kubectl apply -f ./kubernetes/oidc-server-mock.yaml
+}
+
+execute_tests() {
+
+    # running e2e
+    echo "Waiting for things to get ready...($STARTUP_TIMEOUT seconds at most for each service)"
+    kubectl wait pod -l app=oidc-server-mock --for condition=Ready --timeout="$STARTUP_TIMEOUT"s
+
+    echo "Running e2e tests job..."
+    kubectl apply -f ./kubernetes/e2e.yaml
+
+    echo "Waiting for e2e test job to start...($STARTUP_TIMEOUT seconds at most)"
+    kubectl  wait pod -l job-name=e2e --for condition=Ready --timeout "$STARTUP_TIMEOUT"s
+    echo "E2E job is running..."
+    kubectl  logs -l job-name=e2e -f
+    echo "Waiting for e2e test job to complete...($TEST_TIMEOUT seconds at most)"
+    kubectl  wait --for condition=complete --timeout "$TEST_TIMEOUT"s jobs/e2e
+    exitCode=$(kubectl  get pods -l job-name=e2e -o jsonpath='{.items[*].status.containerStatuses[*].state.terminated.exitCode}')
+    reason=$(kubectl  get pods -l job-name=e2e -o jsonpath='{.items[*].status.containerStatuses[*].state.terminated.Reason}')
+
+    echo "The tests finished with exit code $exitCode and message: $reason"
+
+    KUBECONFIG="$oldKUBECONFIG"
+    export KUBECONFIG
+    exit $exitCode
+}
+
+parse_options() {
+    # parse command line options
+    for i in "$@"
+    do
+    case $i in
+        --kind-version=*)
+        KIND_VERSION="${i#*=}"
+        shift
+        ;;
+
+        --kubectl-version=*)
+        KUBECTL_VERSION="${i#*=}"
+        shift
+        ;;
+
+        --cluster-name=*)
+        CLUSTER_NAME="${i#*=}"
+        shift
+        ;;
+
+        --startup-timeout=*)
+        STARTUP_TIMEOUT="${i#*=}"
+        shift
+        ;;
+
+        --test-time=*)
+        TEST_TIMEOUT="${i#*=}"
+        shift
+        ;;
+
+        *)
+        echo "Unknown argument $i"
+        exit 1
+        ;;
+    esac
+    done
+
+    if [[ -z "$KIND_VERSION" ]]
+    then
+        KIND_VERSION="0.3.0"
+    fi
+
+    if [[ -z "$KUBECTL_VERSION" ]]
+    then
+        KUBECTL_VERSION="1.14.3"
+    fi
+
+    if [[ -z "$CLUSTER_NAME" ]]
+    then
+        CLUSTER_NAME="e2e"
+    fi
+
+    if [[ -z "$STARTUP_TIMEOUT" ]]
+    then
+        STARTUP_TIMEOUT=60
+    fi
+
+    if [[ -z "$TEST_TIMEOUT" ]]
+    then
+        TEST_TIMEOUT=60
+    fi
+}
+
+main() {
+    #trap delete_cluster EXIT
+    trap 'report_error_and_exit $LINENO $?' ERR
+    parse_options
```

Comment:
Needs to be `parse_options "$@"`, currently it does not get the parameters and will always use the defaults